src/timer.cr

```diff
@@ -0,0 +1,28 @@
+module SDL
+  module Timer
+    
+    def self.add(interval, &callback : UInt32 -> Void)
+      LibSDL.add_timer(interval, callback)
```

Comment:
I'm afraid `callback` will be run from an unknown context, possibly another thread (unknown to crystal, thus unknown to the garbage collector, and incapable to access the event loop).

It also relies on a captured block passed to a C function, which will make it impossible to rely on closure variables, and there may be GC issues (e.g. callback trying to access collected variables) making it complex to handle, over a simple crystal function, such as `spawn { sleep(seconds), yield }` for example.